rust/src/types.rs

```diff
@@ -938,6 +938,128 @@ impl Type {
         let mut name = QualifiedName::from(name);
         unsafe { BnStr::from_raw(BNGenerateAutoDemangledTypeId(&mut name.0)) }
     }
+
+    pub fn demangle_gnu3<Mangled: BnStrCompatible>(
+        arch: &CoreArchitecture,
+        mangled_name: Mangled,
```

Comment:
I'm not sure it's a good convention, but we use `S` for all our `BnStrCompatible`'s.  Until/unless an effort is made to name those better, I'd prefer to stay consistent in the meantime.

Reply:
Ok I will change that here. Is it worth me trying to give all generics better names in another MR?

Reply:
I'm gonna say no, based on https://doc.rust-lang.org/book/ch10-01-syntax.html:

> You can use any identifier as a type parameter name. But we’ll use T because, by convention, parameter names in Rust are short, often just a letter, and Rust’s type-naming convention is CamelCase. Short for “type,” T is the default choice of most Rust programmers.

and https://rust-lang.github.io/api-guidelines/naming.html:

> Type parameters | concise UpperCamelCase, usually single uppercase letter: T